packages/portfolio-contract/src/plan-solve.ts

```diff
@@ -426,36 +441,80 @@ export const rebalanceMinCostFlowSteps = (
     lastChain = chosen.srcChain;
   }
 
-  const steps: MovementDesc[] = prioritized.map(({ edge, flow }) => {
-    Number.isSafeInteger(flow) ||
-      Fail`flow ${flow} for edge ${edge} is not a safe integer`;
-    const amount = AmountMath.make(graph.brand, BigInt(flow));
-
-    let details = {};
-    switch (edge.feeMode) {
-      case 'gmpTransfer':
-        // TODO: Rather than hard-code, derive from Axelar `estimateGasFee`.
-        // https://docs.axelar.dev/dev/axelarjs-sdk/axelar-query-api#estimategasfee
-        details = { fee: AmountMath.make(graph.feeBrand, 30_000_000n) };
-        break;
-      case 'gmpCall':
-        // TODO: Rather than hard-code, derive from Axelar `estimateGasFee`.
-        // https://docs.axelar.dev/dev/axelarjs-sdk/axelar-query-api#estimategasfee
-        details = { fee: AmountMath.make(graph.feeBrand, 30_000_000n) };
-        break;
-      case 'toUSDN': {
-        // NOTE USDN transfer incurs a fee on output amount in basis points
-        const usdnOut =
-          (BigInt(flow) * (10000n - BigInt(edge.variableFee))) / 10000n;
-        details = { detail: { usdnOut } };
-        break;
+  const padding = 3n; // 300% padding on gas estimates
+  const steps: MovementDesc[] = await Promise.all(
+    prioritized.map(async ({ edge, flow }) => {
+      Number.isSafeInteger(flow) ||
+        Fail`flow ${flow} for edge ${edge} is not a safe integer`;
+      const amount = AmountMath.make(graph.brand, BigInt(flow));
+
+      await null;
+      let details = {};
+      switch (edge.feeMode) {
+        case 'makeEvmAccount':
+          details = {
+            // XXX: hardcoding until we can verify axelar API is accurate for this
+            detail: { evmGas: 200_000_000_000_000n },
+            fee: AmountMath.make(
+              graph.feeBrand,
+              gasEstimator
+                ? (await gasEstimator.getFactoryContractEstimate(
+                    chainOf(edge.dest) as AxelarChain,
+                  )) * padding
+                : 30_000_000n,
+            ),
+          };
```

Comment:
`gasEstimator` should be required.
```suggestion
          const estimate = await gasEstimator.getFactoryContractEstimate(
            chainOf(edge.dest) as AxelarChain,
          );
          // XXX: hardcoding until we can verify axelar API is accurate for this
          const evmGas = 200_000_000_000_000n;
          details = {
            detail: { evmGas },
            fee: AmountMath.make(graph.feeBrand, padFeeEstimate(estimate)),
          };
```